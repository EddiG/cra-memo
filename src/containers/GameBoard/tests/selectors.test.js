import {
  selectCardsQuantity,
  selectFrontImages,
  selectBackImages,
} from '../selectors';

const state = {
  frontImages: {
    byId: {
      frontImage1: {
        id: 'frontImage1',
        src: 'source1',
      },
      frontImage2: {
        id: 'frontImage2',
        src: 'source2',
      },
    },
    allIds: ['frontImage1', 'frontImage2'],
  },
  backImages: {
    byId: {
      backImage1: {
        id: 'backImage1',
        src: 'source1',
      },
      backImage2: {
        id: 'backImage2',
        src: 'source2',
      },
    },
    allIds: ['backImage1', 'backImage2'],
  },
  settings: {
    cardsQuantity: 4,
  },
};

describe('GameBoard selector', () => {
  describe('selectCardsQuantity', () => {
    it('should return cards quantity from the store', () => {
      expect(selectCardsQuantity(state)).toEqual(4);
    });
  });
  describe('selectFrontImages', () => {
    it('should return front images from the store', () => {
      expect(selectFrontImages(state)).toMatchSnapshot();
    });
  });
  describe('selectBackImages', () => {
    it('should return back images from the store', () => {
      expect(selectBackImages(state)).toMatchSnapshot();
    });
  });
});
