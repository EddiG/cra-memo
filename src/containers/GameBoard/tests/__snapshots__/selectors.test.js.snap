// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameBoard selector selectBackImages should return back images from the store 1`] = `
Array [
  "backImage1",
  "backImage2",
]
`;

exports[`GameBoard selector selectFrontImages should return front images from the store 1`] = `
Array [
  "frontImage1",
  "frontImage2",
]
`;
